plugins/flespiConnector.ts

```diff
@@ -112,24 +145,71 @@ function createClient (): MqttClient {
 /**
  * Convert the position string to an actual object
  */
-function convertToObject (json: string): Position {
+function convertToObject(json: string): Position {
     return JSON.parse(json);
 }
 
 /**
  * Emits the new coordinates to the parent component
  */
-function emitNewCoordinates (position: Position): void {
+function emitNewCoordinates(position: Position): void {
     eventBus.$emit("newCoordinates", [position.longitude, position.latitude]);
 }
 
 /**
- * Mqtt method plugin
+ * Calls Flespi API to get the list of all active channels
  */
-const mqttPlugin: Plugin = (_context: Context, inject: Inject) => {
-    inject("getPositionData", () => {
-        return createClient();
-    });
+async function callAPI(): Promise<Channel[]> {
+    // Token needed to authenticate in Flespi
+    const token: string = "FlespiToken " + process.env.FLESPI_KEY;
+    const channels: Channel[] = [];
+
+    // Get the data via the API
+    const pendingResponse = await fetch(
+        "https://flespi.io/gw/channels/all",
+        {
+            headers: {
+                Authorization: token
+            }
+        }
+    );
+
+    const json = await pendingResponse.json();
+
+    for (let i = 0; i < json.result.length; i++) {
+        channels.push(json.result[i]);
+    }
+
+    return channels;
+}
+
+function handleNewPosition(client: MqttClient, result: GeolocationPosition): void {
+    const data = {
+        ident: client.options.clientId || generateRandomId(10),
+        timestamp: result.timestamp / 1000,
+        "position.latitude": result.coords.latitude,
+        "position.longitude": result.coords.longitude,
+        "position.altitude": result.coords.altitude
+    };
```

Comment:
This syntax means that position is an object!
This needs to look something like this:
```suggestion
const data: LocationData = {
    ident: client.options.clientId || generateRandomId(10),
    timestamp: result.timestamp / 1000,
    position: {
        longitude: result.coords.latitude,
        latitude: result.coords.longitude,
        altitude: result.coords.altitude | null
    }
}
```

Reply:
Again, it is to please flespi, I would prefer the object notation aswell